### DISCUSSION TITLE
Deploying Laravel 11 and Filament 3 admin dashboard to shared hosting

### DISCUSSION THREAD

**Discussion by HassanRamos** [filamentphp/filament#13343] [Help]
### Package

Panel builder

### Package Version

3.2

### How can we help you?

I wanted a simple admin dashboard with a blog posting feature, I chose to use filament panel builder, created user admin, and used the FBlog plugin, the admin dashboard is working fine on the local host, deploying on shared hosting, on subdomain, first am having issue am having public in my URL, my subdomain.com/public/admin/login/, when login in am getting The POST method is not supported for route admin/login. Supported methods: GET, HEAD.

**Comment by leandrocfe** [comment]
I think you should map your root folder to `public`, removing `/public/` from url

**Comment by krekas** [comment]
Ditch shared hosting for laravel. If you don't want well then you should find solutions yourself...might sound harsh but with shared it's just how it is

**Comment by HassanRamos** [comment]
a was able to solve by, moving all the content of public folder into root
folder

**Reply by RAFIFEFZET** [reply]
hello I'm also deploying my project I have a problem with the filament CSS not loading because I haven't done npm run build, did you do npm run build because I got an error when npm run build

**Reply by Jesse-Atigedewe** [reply]
same here,figured it out?

**Comment by GrondoyPeru** [comment]
I think they are recommending bad practices by moving everything to the public since it could expose sensitive data,
The best way I have found is 
changing the name of the public folder in Bootstrap/app.php

return Application::configure(basePath: dirname(__DIR__))
    ->registered(function($app){
        $app->usePublicPath($app->basePath("public_html"));
    })
    
Where public_html can be another name
    
That should be enough, also change in vite.config
                publicDirectory: "public_html",
                
and placing everything so that only public goes to public_html
So that it is as follows
![imagen](https://github.com/user-attachments/assets/66d97b86-f384-4605-824e-36158b4585e3)
